### ISSUE TITLE
`PwParser`: Add atomic charges and magnetic moments to `output_parameters`

### ISSUE THREAD

**Issue by mbercx** [aiidateam/aiida-quantumespresso#734]
While the current `output_parameters` _do_ contain the `atomic_charges_units` and `atomic_magnetic_moments_units`, they do not actually contain the final `atomic_charges` and `atomic_magnetic_moments` themselves. There the user has to find in the `output_trajectory`, but that makes them somewhat hard to find. Since these are quite critical outputs, I suggest we add the final charges and magnetic moments to the `output_parameters` as well.

**Comment by sphuber** [comment]
That would mean we would be duplicating the information, wouldn't we? Instead of adding more to the `output_parameters` would it make sense to just document this properly and eventually maybe remove `atomic_charges_units` and `atomic_magnetic_moments_units`.

**Comment by mbercx** [comment]
True, but is the duplication such a big issue? Documenting this is definitely a good idea, but users don't always find their way to the correct part of the documentation easily. I think users will naturally look for these values in the `output_parameters`, at least I've tried to find them there multiple times. 😁

**Comment by sphuber** [comment]
Duplication can be a big issue if the values are relatively large and can cause database bloat. It is always a trade-off, but I wanted to put the argument here so we consider it.

**Comment by mbercx** [comment]
Of course, fair. I would say it's acceptable. Note that I would only add the _final_ charges and magnetic moments, so this would be `len(StructureData.sites)` values for each. It's in a sense similar to what we were considering regarding the magnetic moments to the `StructureData` type by creating a new `MagneticStructureData`, for example.

**Comment by sphuber** [comment]
Just to provide a quick estimate. This would mean adding `2 * 8 * len(structure.sites)` bytes per `StructureData`. If we assume an average of 50 atoms for the 3DCD (not sure if that is accurate but just for arguments sake) we would be adding `2 * 8 * 50 * 400,000` equals 320 MB. That is just for the base structures and not any of the relaxed ones. Of course this is for one of the bigger databases, but I do think it gives a sense of how much we are adding that is purely duplicate info. But if you really think it is not findable in the `output_trajectory`, then we can add it.

**Comment by mbercx** [comment]
Hmm, I see more and more the trade-off. 🙃  I just checked, and the database of the 3DCD is 40G. So based on your estimate we'd be adding close to 1% more size to the database, which is not trivial. What are the consequences of a larger database (in size, not number of nodes stored) besides just requiring more storage (and slower backups etc). Is it more sluggish?

**Comment by sphuber** [comment]
> What are the consequences of a larger database (in size, not number of nodes stored) besides just requiring more storage (and slower backups etc). Is it more sluggish?

Given that this data would be stored on the `attributes` column of the node table, it is very possible that a good amount of queries that involve that table and/or column would be slowed down by additional content. I can't really give an estimate of the amount to be honest. Could be little, could be a lot, depending on the query. But since it can have an effect, it is good to be aware of what we store in the database.
